### ISSUE TITLE
Not able to edit code and answers after submission closes

### ISSUE THREAD

**Issue by Joulenergy** [source-academy/frontend#3074]

**Comment by sayomaki** [comment]
Would you mind elaborating on this? When the submission closes, no more changes can be made and that is the intended behaviour.

**Comment by Joulenergy** [comment]
Yes! I mean that it will be useful if we have a feature that allows us to make changes to the submitted answers for revision. 
For example if the path was not submitted, everything will be red right, but if the student wants to revise, it would help if they can try again and then save the green version for themselves so when they revise they do not have to keep clicking again to check which one is the correct one and be unable to save. 
Same goes for the code, for example in the path they have the mini coding questions that need to pass the test cases, they can try to write it after submission and run the test cases, but they have no way to save the code that they have edited

**Comment by martin-henz** [comment]
I think the more general topic is "versioning". At the moment, we only keep one copy of student submissions. It would be GREAT if we would keep all saved versions with time stamps. This would by itself be a great CP3108 project, involving frontend and backend.

**Comment by geraldnyeo** [comment]
### [Proposal] CP3108 26MH: P9 Versioning and history
**Overview**
Currently, only the latest copy of the student’s code is saved. We would like to propose a design for the versioning and history system, similar to that of Google Docs.

**Features**
We would like to propose the following features:
1. Undo/Redo Buttons on the control bar
2. Version History View
3. Ability to restore previously saved versions
4. Ability to name previously saved versions
5. Autosaving of the student’s code

**Mockups**
*Control Bar*
<img width="1270" height="84" alt="Image" src="https://github.com/user-attachments/assets/7082ad87-b994-4062-9060-9f27b0e11e62" />

*Version History Panel*
<img width="1268" height="502" alt="Image" src="https://github.com/user-attachments/assets/ad50f649-e62c-4772-a6c2-5a5926997280" />


*Version History Panel - preview previous version*
<img width="1270" height="504" alt="Image" src="https://github.com/user-attachments/assets/a03ad0e0-2234-456f-a82e-bacd247578d1" />


*Version History Panel - restore previous version*
<img width="1266" height="498" alt="Image" src="https://github.com/user-attachments/assets/0da67e40-a0f6-4500-92a2-45dd0a9386a7" />
